cmake/CodeFormat.cmake

```diff
@@ -0,0 +1,51 @@
+##---------------------------------------------------------------------------##
+## Check C++ code style using clang-format version 3.9
+##---------------------------------------------------------------------------##
+
+if(DataTransferKit_ENABLE_ClangFormat)
+    find_program(CLANG_FORMAT_EXECUTABLE clang-format-3.9
+        NAMES clang-format-mp-3.9 # on OS X when installed using MacPorts
+    )
+    if(CLANG_FORMAT_EXECUTABLE)
+        message("-- Found clang-format: ${CLANG_FORMAT_EXECUTABLE}")
+    else()
+        message(FATAL_ERROR "-- clang-format not found")
+    endif()
+    file(DOWNLOAD
+        https://raw.githubusercontent.com/ORNL-CEES/Cap/master/diff-clang-format.py
+        ${CMAKE_BINARY_DIR}/diff-clang-format.py
+    )
+    add_custom_target(format-cpp
+        ${PYTHON_EXECUTABLE} ${CMAKE_BINARY_DIR}/diff-clang-format.py
+            --file-extension='.hpp'
+            --file-extension='.cpp'
+            --style=file
+            --config=${${PACKAGE_NAME}_SOURCE_DIR}/.clang-format
+            --apply-patch
+            ${${PACKAGE_NAME}_SOURCE_DIR}/packages
+    )
+    file(WRITE
+        ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/check_format_cpp.sh
+        "#!/usr/bin/env bash\n"
+        "\n"
+        "${PYTHON_EXECUTABLE} "
+        "${CMAKE_BINARY_DIR}/diff-clang-format.py "
```

Comment:
diff-clang-format.py seems to produce counter-intuitive diff. It seems to first print the new version with <, and then the old version with >. This is not how the regular diff works. Instead, it should first print old version with <, and then the new version with >.